x/app-router/keeper/liquidstake_test.go

```diff
@@ -0,0 +1,155 @@
+package keeper_test
+
+import (
+	"fmt"
+	"time"
+
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/Stride-Labs/stride/v3/x/app-router/types"
+	epochtypes "github.com/Stride-Labs/stride/v3/x/epochs/types"
+	minttypes "github.com/Stride-Labs/stride/v3/x/mint/types"
+	recordstypes "github.com/Stride-Labs/stride/v3/x/records/types"
+	stakeibctypes "github.com/Stride-Labs/stride/v3/x/stakeibc/types"
+)
+
+func (suite *KeeperTestSuite) TestTryLiquidStaking() {
+	now := time.Now()
+
+	packet := channeltypes.Packet{
+		Sequence:           1,
+		SourcePort:         "transfer",
+		SourceChannel:      "channel-0",
+		DestinationPort:    "transfer",
+		DestinationChannel: "channel-0",
+		Data:               []byte{},
+		TimeoutHeight:      clienttypes.Height{},
+		TimeoutTimestamp:   0,
+	}
+
+	atomHostDenom := "uatom"
+	prefixedDenom := transfertypes.GetPrefixedDenom(packet.GetDestPort(), packet.GetDestChannel(), atomHostDenom)
+	atomIbcDenom := transfertypes.ParseDenomTrace(prefixedDenom).IBCDenom()
+
+	strdDenom := "ustrd"
+	prefixedDenom = transfertypes.GetPrefixedDenom(packet.GetSourcePort(), packet.GetSourceChannel(), strdDenom)
+	strdIbcDenom := transfertypes.ParseDenomTrace(prefixedDenom).IBCDenom()
+
+	testCases := []struct {
+		forwardingActive bool
+		recvDenom        string
+		packetData       transfertypes.FungibleTokenPacketData
+		expNilAck        bool
+	}{
+		{ // params not enabled
```

Comment:
Can we add an additional failure case where the receiver address is invalid, and ensure that the tokens are refunded?

Reply:
Okay

Reply:
TryLiquidStaking codebase does not have function to check address, since address is passed in AccAddress, and parse is done on parseReceiverAddress.

Reply:
Added this commit for checking additional failure case when receiver address is invalid
https://github.com/Stride-Labs/stride/pull/432/commits/9e41f220f65f9c899b56edd15e57027ed57d5a35